prac_08/squaring.py

```
@@ -0,0 +1,35 @@
+"""
+CP1404/CP5632 Practical
+Kivy GUI program to square a number
+Lindsay Ward, IT@JCU
+Started 13/10/2015
+
+Modifications by Hamzah "Opal" Nutt
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Lindsay Ward, modified by Hamzah "Opal" Nutt'
+
+
+class SquareNumberApp(App):
+    """SquareNumberApp is a Kivy App for squaring a number."""
+    def build(self):
+        """Build the Kivy app from the kv file."""
+        Window.size = (200, 100)
+        self.title = "Square Number"
+        self.root = Builder.load_file('squaring.kv')
+        return self.root
+
+    def handle_calculate(self, value):
+        """Handle calculation (could be button press or other call), output result to label widget."""
+        try:
+            result = float(value) ** 2
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
+            pass
```

The except ValueError block silently passes. Add user feedback (e.g., updating the label to show an error message like "Invalid input").

---

Possible, but the question didn't ask for it. The `except ValueError` block is unmodified from the demo file we were asked to use.